src/DaiDripsHub.sol

```diff
@@ -84,7 +84,7 @@ contract DaiDripsHub is ERC20DripsHub {
     /// These parameters will be passed to the Dai contract by this function.
     /// @param permitArgs The Dai permission arguments.
     function giveAndPermit(
-        address receiver,
+        uint256 receiver,
```

Comment:
One idea to increase the readability of the code can be to introduce a struct with only one uint256 as member for UserIds.

```solidity
struct UserId {
uint256 value;
}
```

The gas cost would be the same because in both cases it is just a `uint` just some syntax sugar for solidity, but it would allow 

```solidity
function functionName(UserId receiver,...)
```

Reply:
Maybe even better, if we want to implement it. Could be still an overkill.

```
// struct
uint4 accountId;
uint28 subAccountId;
```

In that case we would avoid the bit manipulation. You could just access the `subAccountId`.